processing/src/main/java/io/druid/segment/filter/AndFilter.java

```diff
@@ -71,6 +77,68 @@ public ValueMatcher makeMatcher(ValueMatcherFactory factory)
     return makeMatcher(matchers);
   }
 
+  @Override
+  public ValueMatcher makeMatcher(
+      BitmapIndexSelector selector,
+      ValueMatcherFactory valueMatcherFactory,
+      RowOffsetMatcherFactory rowOffsetMatcherFactory
+  )
+  {
+    final List<ValueMatcher> offsetMatchers = new ArrayList<>();
+    final List<ValueMatcher> valueMatchers = new ArrayList<>();
+
+    for (Filter filter : filters) {
+      ValueMatcher matcher = filter.makeMatcher(selector, valueMatcherFactory, rowOffsetMatcherFactory);
+      if (filter.supportsBitmapIndex(selector)) {
+        offsetMatchers.add(matcher);
+      } else {
+        valueMatchers.add(matcher);
+      }
+    }
+
+    return new ValueMatcher()
```

Comment:
Additionally a comment about how one day the filters that support bitmap indexes could be getBitmapIndex'ed rather than makeMatcher'ed (or just do that)